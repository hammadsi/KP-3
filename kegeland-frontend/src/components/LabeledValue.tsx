import { HStack, Icon, VStack, Text } from '@chakra-ui/react';
import { ComponentProps } from 'react';
import { IconType } from 'react-icons/lib';

type LabeledValueProps = {
  label: string;
  value: string | number;
  icon?: IconType;
  iconProps?: ComponentProps<typeof Icon>;
} & ComponentProps<typeof HStack>;

const LabeledValue: React.FC<LabeledValueProps> = ({
  label,
  value,
  icon,
  iconProps,
  ...props
}) => {
  return (
    <HStack {...props}>
      <VStack alignItems="flex-start" spacing={0}>
        <Text fontWeight="semibold" color="gray.800">
          {label}:
        </Text>
        <HStack spacing={0.5}>
          {icon && (
            <Icon as={icon} {...iconProps} color="primary.700" fontSize={18} />
          )}
          <Text marginBlockStart={0} marginBlockEnd={0} fontSize={14}>
            {value}
          </Text>
        </HStack>
      </VStack>
    </HStack>
  );
};

export default LabeledValue;
