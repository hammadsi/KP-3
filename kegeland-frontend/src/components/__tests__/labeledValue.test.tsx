import renderer from 'react-test-renderer';

import LabeledValue from '../LabeledValue';

describe('Test labeled value', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<LabeledValue label={'label'} value={123} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
