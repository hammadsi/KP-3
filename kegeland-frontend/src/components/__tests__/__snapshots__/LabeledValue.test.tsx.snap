// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test labeled value renders correctly 1`] = `
<div
  className="chakra-stack css-84zodg"
>
  <div
    className="chakra-stack css-x09s84"
  >
    <p
      className="chakra-text css-1lgbcr5"
    >
      label
      :
    </p>
    <div
      className="chakra-stack css-64a7jp"
    >
      <p
        className="chakra-text css-16h254d"
      >
        123
      </p>
    </div>
  </div>
</div>
`;
